packages/portfolio-contract/src/portfolio.exo.ts

```diff
@@ -39,195 +45,187 @@ export const DECODE_CONTRACT_CALL_RESULT_ABI = [
-            } else {
-              throw makeError(
-                `no matching chainId ${targetChainId}: ${q({ Aave: Aave?.chain, Compound: Compound?.chain })}`,
+            if (message.success) {
+              const [address] = decodeAbiParameters(
+                [{ type: 'address' }],
+                message.result,
               );
+              const { Gmp } = this.state;
+              if (Gmp) {
+                // TODO: remoteAddressVK is set only once
+                Gmp.remoteAddressVK.resolver.resolve(address);
+              }
+              trace(`remoteAccountAddress ${address}`);
             }
           }
-
+          // TODO: Handle the result of the contract call
           trace('receiveUpcall completed');
         },
       },
       keeper: {
         getLCA() {
-          const { localAccount } = this.state;
-          return localAccount;
+          const { Gmp } = this.state;
+          if (!Gmp) throw Error('Gmp not set');
+          return Gmp.localAccount;
         },
         getPositions(): YieldProtocol[] {
           const { state } = this;
           return harden(
             (keys(YieldProtocol) as YieldProtocol[]).filter(k => !!state[k]),
           );
         },
+        getUSDNICA() {
+          const { USDN } = this.state;
+          if (!USDN) throw Error('USDN not set');
+          return USDN;
+        },
+        getRemoteAddress() {
+          const { Gmp } = this.state;
+          if (!Gmp) throw Error('Gmp not set');
+          return Gmp.remoteAddressVK.vow;
+        },
         initAave(chain: CaipChainId) {
           this.state.Aave = harden({
             chain,
-            remoteAddressVK: vowTools.makeVowKit(),
           });
-          trace('initAave', chain);
         },
-        initCompound() {
+        initCompound(chain: CaipChainId) {
           this.state.Compound = harden({
-            chain: EthL1,
-            remoteAddressVK: vowTools.makeVowKit(),
+            chain,
           });
-          trace('initCompound');
         },
         initUSDN(account: NobleAccount) {
           this.state.USDN = account;
-          trace('initUSDN', account);
         },
-        getUSDNICA() {
-          const { USDN } = this.state;
-          if (!USDN) throw Error('USDN not set');
-          return USDN;
-        },
-        getAaveAddress() {
-          const { Aave } = this.state;
-          if (!Aave) throw Error('Aave not set');
-          return Aave.remoteAddressVK.vow;
+
+        initGmp(account: LocalAccount) {
+          this.state.Gmp = harden({
+            localAccount: account,
+            remoteAddressVK: vowTools.makeVowKit(),
+          });
+          trace('initLCA');
         },
-        getCompoundAddress() {
-          const { Compound } = this.state;
-          if (!Compound) throw Error('Compound not set');
-          return Compound.remoteAddressVK.vow;
+        wait(val: bigint) {
+          return vowTools.watch(E(timer).delay(val));
```

Comment:
I don't think `watch` works to make this sort of thing survive upgrade. I think you have to use `retryable`, right @michaelfig ?

But given that this is a short-term kludge, let's just mark it as such:

```suggestion
        /** KLUDGE around lack of synchronization signals for now. TODO: rethink design. */
        wait(val: bigint) {
          return vowTools.watch(E(timer).delay(val));
```